cmd/collectors/zapi/collector/zapi.go

```diff
@@ -147,6 +148,12 @@ func (z *Zapi) InitVars() error {
 	if timeout := z.Params.GetChildContentS("client_timeout"); timeout != "" {
 		z.Client.SetTimeout(timeout)
 	}
+
+	// if the object template includes a run_plugins_if_no_data, then honour that
+	z.RunPluginsIfNoData = false
```

Comment:
z.RunPluginsIfNoData = false isn't needed since it will default to false
you could simplify these lines to

```go
z.RunPluginsIfNoData = z.Params.HasChildS("run_plugins_if_no_data")
```

but the `if` version is fine too if you prefer that one for readability.

Reply:
Done